library/Icingadb/Widget/ItemList/BaseHostListItem.php

```diff
@@ -42,6 +42,7 @@ protected function init()
 
         if ($this->list->getNoSubjectLink()) {
             $this->setNoSubjectLink();
+            $this->isDetailView = true;
```

Comment:
I am not sure if `getNoSubjectLink()` always means that it is a detail view. Maybe it would be better to add a method like isDetailView().

```
$this->addControl((new HostList([$this->host]))
            ->setViewMode('minimal')
            ->setDetailActionsDisabled()
            ->setNoSubjectLink()
            ->isDetailView());
```

Reply:
You're correct with the assumption that `getNoSubjectLink()` is no reliable indicator if it's a detail view where this is shown.

But another modifier is not desirable in my opinion. I'm already no real fan of the fact that the detail view uses a state list in the header, that should be a dedicated widget. Then we wouldn't have this discussion or any of the existing modifiers.

Though, creating the widget here is again out of scope, but still, I dislike an additional modifier. Use a new view mode that inherits from the minimal one instead. There override `getStateBallSize()` to choose the proper size and also override `assembleVisual` where in you call the super method first and then fetch the state ball using `HtmlDocument::first()` and register the div with the previous state as wrapper for it.